### PR TITLE
Skip apply annotation when copying annotations from deployment to RC

### PR DESCRIPTION
**Pull request by janetkuo**
To address https://github.com/kubernetes/kubernetes/pull/20035#issuecomment-179558615:

> Copying the apply annotations will interfere with the culling proposal: #19805. So, we shouldn't copy them, but that raises the question of how to decide which to copy and which not to copy. I don't really want to make that configurable via the API and I don't want an ever-expanding hardcoded whitelist of annotations to copy, either. So, maybe for 1.2 we should just avoid copying the apply annotation, wait for user feedback on the issue, and decide whether to change the policy in 1.3.

cc @bgrant0607 @nikhiljindal @ironcladlou @kargakis @kubernetes/sig-config

### CODE REVIEW

File: pkg/controller/deployment/deployment_controller.go [hunk 1]
```diff
@@ -675,24 +676,8 @@ func (dc *DeploymentController) getNewRC(deployment extensions.Deployment, maxOl
...
-		}
-		// Update existing new RC's revision annotation
-		if existingNewRC.Annotations[deploymentutil.RevisionAnnotation] != newRevision {
-			existingNewRC.Annotations[deploymentutil.RevisionAnnotation] = newRevision
-			annotationChanged = true
-			glog.V(4).Infof("update existingNewRC %s revision to %s - %+v\n", existingNewRC.Name, newRevision)
-		}
-		if annotationChanged {
+		// Set existing new RC's annotation
+		if setNewRCAnnotaions(&deployment, existingNewRC, newRevision) {
```
Comment (bgrant0607) [inline pkg/controller/deployment/deployment_controller.go]: Typo: Annotations

**Comment by bgrant0607** [comment]
LGTM, thanks

File: pkg/controller/deployment/deployment_controller.go [hunk 2]
```diff
@@ -750,6 +728,43 @@ func (dc *DeploymentController) getNewRC(deployment extensions.Deployment, maxOl
...
+	}
+	return annotationChanged
+}
+
+// copyDeploymentAnnotationsToRC copies deployment's annotations to rc's annotations,
+// and returns true if rc's annotation is changed
+func copyDeploymentAnnotationsToRC(deployment *extensions.Deployment, rc *api.ReplicationController) bool {
+	rcAnnotationsChanged := false
+	for k, v := range deployment.Annotations {
+		if rc.Annotations == nil {
```
Comment (kargakis) [inline pkg/controller/deployment/deployment_controller.go]: move this check out of the loop

Comment (janetkuo) [inline pkg/controller/deployment/deployment_controller.go]: Done

File: pkg/controller/deployment/deployment_controller.go (hunk 1 above)
Comment (janetkuo) [inline pkg/controller/deployment/deployment_controller.go]: Fixed

**Comment by kargakis** [comment]
LGTM

**Comment by janetkuo** [comment]
@k8s-bot e2e test this

**Pull request merged**